### DISCUSSION TITLE
`deep` in the js SDK

### DISCUSSION THREAD

**Discussion by cupcakearmy** [directus/directus#3424] [Archive]
Hi there,

I've seen there is a `deep` attribute to the `Query` object. 
Could not find any docs to it.
Guessing from the types i tried stuff like `{ deep: { foo: { fields: '*.*' } } }`
Where `foo` would be a relation in the collection type.

Is it something to do with relations? Would be awesome as the normal 'fields' can get quite big xD

**Comment by rijkvanzanten** [comment]
You're right!

`deep` can be used to apply any of the other query params to nested data sets, so you can set it like

```
deep: {
  favorites: {
    limit: 5
  }
}
```

That being said

> Would be awesome as the normal 'fields' can get quite big

You can also specify fields as an array instead of a string if you don't do that already:

```
fields: [
  'id',
  'author.name',
  'etc
]
```

which helps with the code organization

**Reply by cupcakearmy** [reply]
Thanks for the fast reply. This is really useful! Thanks for the clarification :)

**Reply by IamSAB** [reply]
Somehow that does not work for the /items api.

In a query `fields=children.title` works, but `deep[children][fields]=title` not whereas children is a relational field. I also tried it with `deep={"children": {"fields": "title"}}` which isn't working too.

What am I doing wrong?

**Reply by altezza04** [reply]
> Somehow that does not work for the /items api.
> 
> In a query `fields=children.title` works, but `deep[children][fields]=title` not whereas children is a relational field. I also tried it with `deep={"children": {"fields": "title"}}` which isn't working too.
> 
> What am I doing wrong?

I experience the same, this work in API"

`?deep[children][filter][status][_eq]=published`

but it doesnt give filter the status in SDK if i do this

`          deep: {
            children: {
              filter: {
                status: { _eq: 'published' }
              }
            }
          }
`

**Reply by lessborders** [reply]
This still doesnt work in the SDK??

**Reply by rijkvanzanten** [reply]
> This still doesnt work in the SDK??

Yes it does. The SDK just passes all the query params to the API, so the full API functionality is available from the SDK as well 👍🏻 That being said, lets keep that discussion going on https://github.com/directus/sdk/issues/55

**Reply by notflip** [reply]
@rijkvanzanten Thanks for your answer, how do you target a deeper relation?
For example if I have `blocks.item.translations.*`, how can I filter those translations using deep?

This doesn't seem to work
```
    deep: [
      { translations: { '_filter': { 'languages_id': { '_eq': 'en-US' } } } }
    ],
```

Or this 
```
deep: [
  {
    blocks: {
      item: {
        translations: { '_filter': { 'languages_id': { '_eq': 'en-US' } } }
      }
    }
  }
],
```
